code-analysis/crates/solidity/outputs/rust/lib/src/generated/language.rs

```diff
@@ -19,1041 +23,1087 @@ impl Language {
+        ParserOutput::new(source, &self.parsers.yul_if_statement)
     }
 
     // «YulKeyword» = «BooleanLiteral» | 'break' | 'case' | 'continue' | 'default' | 'for' | 'function' | 'hex' | 'if' | 'leave' | 'let' | 'switch' ;
-    pub fn parse_yul_keyword(&self, source: &str) -> Rc<cst::Node> {
-        let (node, _errs) = self.parsers.yul_keyword.parse_recovery(source);
-        node.unwrap()
+    pub fn parse_yul_keyword(&self, source: &str) -> ParserOutput {
+        ParserOutput::new(source, &self.parsers.yul_keyword)
     }
 
     // YulLeaveStatement = 'leave' ;
-    pub fn parse_yul_leave_statement(&self, source: &str) -> Rc<cst::Node> {
-        let (node, _errs) = self.parsers.yul_leave_statement.parse_recovery(source);
-        node.unwrap()
+    pub fn parse_yul_leave_statement(&self, source: &str) -> ParserOutput {
+        ParserOutput::new(source, &self.parsers.yul_leave_statement)
     }
 
     // YulLiteral = «YulDecimalNumberLiteral» | «YulHexLiteral» | «AsciiStringLiteral» | «BooleanLiteral» | «HexStringLiteral» ;
-    pub fn parse_yul_literal(&self, source: &str) -> Rc<cst::Node> {
-        let (node, _errs) = self.parsers.yul_literal.parse_recovery(source);
-        node.unwrap()
+    pub fn parse_yul_literal(&self, source: &str) -> ParserOutput {
+        ParserOutput::new(source, &self.parsers.yul_literal)
     }
 
     // YulStatement = YulBlock | YulVariableDeclaration | YulFunctionDefinition | YulAssignmentStatement | YulFunctionCall | YulIfStatement | YulForStatement | YulSwitchStatement | YulLeaveStatement | YulBreakStatement | YulContinueStatement ;
-    pub fn parse_yul_statement(&self, source: &str) -> Rc<cst::Node> {
-        let (node, _errs) = self.parsers.yul_statement.parse_recovery(source);
-        node.unwrap()
+    pub fn parse_yul_statement(&self, source: &str) -> ParserOutput {
+        ParserOutput::new(source, &self.parsers.yul_statement)
     }
 
     // YulSwitchStatement = 'switch' YulExpression 1…*{ ( 'case' YulLiteral | 'default' ) YulBlock } ;
-    pub fn parse_yul_switch_statement(&self, source: &str) -> Rc<cst::Node> {
-        let (node, _errs) = self.parsers.yul_switch_statement.parse_recovery(source);
-        node.unwrap()
+    pub fn parse_yul_switch_statement(&self, source: &str) -> ParserOutput {
+        ParserOutput::new(source, &self.parsers.yul_switch_statement)
     }
 
     // YulVariableDeclaration = 'let' YulIdentifierPath  { ',' YulIdentifierPath } [ ':=' YulExpression ] ;
-    pub fn parse_yul_variable_declaration(&self, source: &str) -> Rc<cst::Node> {
-        let (node, _errs) = self.parsers.yul_variable_declaration.parse_recovery(source);
-        node.unwrap()
+    pub fn parse_yul_variable_declaration(&self, source: &str) -> ParserOutput {
+        ParserOutput::new(source, &self.parsers.yul_variable_declaration)
+    }
+
+    pub fn parse(&self, parser_name: &str, source: &str) -> Option<ParserOutput> {
```

Comment:
I suggest we return an `Err()` if the `parser_name` is incorrect instead of `None`, as the latter suggests to me that parsing is successful but didn't produce an output, as oppposed to failing to find a parser to run.

We can also use `this_error` crate and return a specific `ParserNotFound` error enum variant if you like the suggestion.

Reply:
I originally had Result<...> but I thought that analogous to HashMap::get, it would be better to allow the caller to decide if it was an error or not.